# Task 30 — High-Usage Customers by Subscription Plan.

## Business Question

How many customers in each subscription plan use more than 3,000 GB of downloaded data?

## Analysis

Filtered customers with gb_downloaded > 3000, then grouped them by plan_name and counted the number of customers in each subscription plan. The results were sorted in descending order by the number of high-usage customers.

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
users_by_plan_and_more_3000gb = (
    starlink[starlink["gb_downloaded"] > 3000]
    .groupby("plan_name").agg(
        users_amount = ("unique_id","count")
    )
).sort_values(
    "users_amount",
    ascending = False
)
print(users_by_plan_and_more_3000gb)

             users_amount
plan_name                
Residential           793
Roam                  285
Business              146
Mini                  130


### Business Insight

Residential has the highest number of high-usage customers (793), followed by Roam (285), Business (146), and Mini (130). This indicates that Residential customers represent the largest group of customers consuming more than 3,000 GB of downloaded data.